src/core/config/transformer/rename_types.rs

```diff
@@ -65,6 +83,51 @@ impl Transform for RenameTypes {
                         }
                     }
                 }
+
+                // replace in interface.
+                type_.implements = type_
+                    .implements
+                    .iter()
+                    .filter_map(|interface_type_name| {
+                        lookup
+                            .get(interface_type_name)
+                            .cloned()
+                            .or(Some(interface_type_name.clone()))
+                    })
+                    .collect();
+            }
+
+            // replace in the union as well.
+            for union_type_ in config.unions.values_mut() {
+                // Collect changes to be made
+                let mut types_to_remove = HashSet::new();
+                let mut types_to_add = HashSet::new();
+
+                for type_name in union_type_.types.iter() {
+                    if let Some(new_type_name) = lookup.get(type_name) {
+                        types_to_remove.insert(type_name.clone());
+                        types_to_add.insert(new_type_name.clone());
+                    }
+                }
+                // Apply changes
+                for type_name in types_to_remove {
+                    union_type_.types.remove(&type_name);
+                }
+
+                for type_name in types_to_add {
+                    union_type_.types.insert(type_name);
+                }
+            }
+
+            // replace in union as well.
+            for union_type_ in config.unions.values_mut() {
+                union_type_.types = union_type_
+                    .types
+                    .iter()
+                    .filter_map(|type_name| {
```

Comment:
1) why we also use `filter_map`?
2) replace `or` with `or_else` because it lazily evaluates